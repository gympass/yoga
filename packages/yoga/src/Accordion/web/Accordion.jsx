import React, { useState } from 'react';
import styled, { css } from 'styled-components';
import { string, node, bool } from 'prop-types';
import { ChevronDown } from '@gympass/yoga-icons';
import { Text, Divider } from '../..';
import Content from './Content';

const AccordionWrapper = styled.div`
  background-color: ${({ theme }) => theme.yoga.colors.white};
  border: none;
  display: flex;
  flex-direction: column;
  position: relative;
  width: 100%;
  z-index: 1;

  ${({
    disabled,
    theme: {
      yoga: {
        colors: { elements },
      },
    },
  }) =>
    disabled &&
    css`
      background-color: ${elements.backgroundAndDisabled};
    `}

  hr {
    bottom: 0;
    left: 0;
    margin: 0;
    position: absolute;
    z-index: -1;
  }
`;

const getVerticalPadding = (accordion, small) => {
  if (small) return accordion.padding.small;
  return accordion.padding.large;
};

const getHorizontalPadding = (accordion, hasHorizontalPadding) => {
  if (hasHorizontalPadding) return accordion.padding.standard;
  return accordion.padding.zero;
};

const Header = styled.button`
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: flex;
  justify-content: space-between;
  width: 100%;

  ${({
    theme: {
      yoga: {
        components: { accordion },
      },
    },
    small,
    hasHorizontalPadding,
  }) => {
    return `
  padding:${getVerticalPadding(accordion, small)}px ${getHorizontalPadding(
      accordion,
      hasHorizontalPadding,
    )}px;
`;
  }}

  &:focus-visible {
    ${({
      theme: {
        yoga: {
          colors: { elements },
        },
      },
    }) => {
      return `
    background: ${elements.backgroundAndDisabled};

  `;
    }}
  }

  ${({ disabled }) =>
    disabled &&
    css`
      cursor: not-allowed;
    `}
`;

const Title = styled.div`
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  text-align: start;

  ${({
    theme: {
      yoga: {
        components: { accordion },
      },
    },
    subtitle,
  }) => {
    return `
  gap: ${subtitle ? accordion.gap.header : 0}px;
  margin: ${subtitle ? 0 : `${accordion.paddingArrow.total}px 0`};
`;
  }}

  ${({ disabled }) =>
    disabled &&
    css`
      ${({
        theme: {
          yoga: {
            colors: { text },
          },
        },
      }) => {
        return `
      color: ${text.secundary};
    `;
      }}
    `}
`;

const AccordionContent = styled.div`
  height: auto;
  max-height: ${({ isOpen }) => (isOpen ? '9999px' : '0')};
  overflow: hidden;
  transition: max-height 200ms ease-in-out;

  ${({
    theme: {
      yoga: {
        components: { accordion },
      },
    },
    small,
  }) =>
    small &&
    css`
      ${Content} {
        padding-bottom: ${accordion.padding.small}px;
      }
    `}

  ${({ hasHorizontalPadding }) =>
    !hasHorizontalPadding &&
    css`
      ${Content} {
        padding-left: 0;
        padding-right: 0;
      }
    `}
`;

const ArrowWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;

  svg {
    transform: ${({ isOpen }) => (isOpen ? 'rotate(180deg)' : 'rotate(0deg)')};
    transition: all 200ms ease-out 0s;

    ${({
      theme: {
        yoga: {
          colors: { primary, text },
        },
      },
      disabled,
    }) => {
      return `
      fill: ${disabled ? text.secondary : primary};
    `;
    }}
  }

  ${({
    theme: {
      yoga: {
        components: { accordion },
      },
    },
  }) => {
    return `
  padding: ${accordion.paddingArrow.total}px;
`;
  }}
`;

const Accordion = ({
  title,
  subtitle,
  children,
  disabled = false,
  expanded = false,
  small = false,
  hasHorizontalPadding = true,
  ...props
}) => {
  const [open, setOpen] = useState(expanded);

  const [summary, ...contentChildren] = React.Children.toArray(children);

  const hasSummary = summary.type.displayName === 'Accordion.Summary';

  return (
    <AccordionWrapper disabled={disabled} {...props}>
      <Header
        disabled={disabled}
        onClick={() => {
          setOpen(o => !o);
        }}
        small={small}
        hasHorizontalPadding={hasHorizontalPadding}
      >
        {hasSummary ? (
          summary
        ) : (
          <Title subtitle={subtitle}>
            {small ? (
              <Text.Small color={disabled ? 'deep' : undefined}>
                {title}
              </Text.Small>
            ) : (
              <Text.Body1
                bold
                color={disabled ? 'deep' : undefined}
                lineHeight="medium"
              >
                {title}
              </Text.Body1>
            )}

            <Text.Small color="deep">{subtitle}</Text.Small>
          </Title>
        )}

        <ArrowWrapper isOpen={open} disabled={disabled}>
          <ChevronDown width={24} height={24} />
        </ArrowWrapper>
      </Header>

      <AccordionContent
        isOpen={open}
        small={small}
        hasHorizontalPadding={hasHorizontalPadding}
      >
        {hasSummary ? contentChildren : children}
      </AccordionContent>
      <Divider />
    </AccordionWrapper>
  );
};

Accordion.propTypes = {
  title: string,
  subtitle: string,
  children: node.isRequired,
  disabled: bool,
  expanded: bool,
  small: bool,
  hasHorizontalPadding: bool,
};

export default Accordion;
