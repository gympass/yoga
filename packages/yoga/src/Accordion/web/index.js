import Accordion from './Accordion';
import Summary from './Summary';
import Content from './Content';
import AccordionSmall from './Small';

const ExportAccordion = Object.assign(Accordion, {
  Content,
  Summary,
  Small: AccordionSmall,
});

export default ExportAccordion;
