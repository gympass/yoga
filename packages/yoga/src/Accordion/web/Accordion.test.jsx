import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import {
  fontSizes,
  fontWeights,
  spacing,
} from '@gympass/yoga-tokens/src/global';
import { Text, Button, ThemeProvider, v3theme } from '../..';

import Accordion from '.';

describe('<Accordion />', () => {
  afterEach(cleanup);

  describe('should correctly render', () => {
    it('the default version', () => {
      const { container } = render(
        <ThemeProvider>
          <Accordion title="Default" expanded>
            <Accordion.Content>
              <Text>Content</Text>
            </Accordion.Content>
          </Accordion>
        </ThemeProvider>,
      );

      expect(screen.getByText('Default')).toBeInTheDocument();

      expect(screen.getByText('Default')).toHaveStyle({
        fontSize: `${fontSizes.medium}px`,
        fontWeight: fontWeights.medium,
      });

      expect(screen.getByRole('button')).toHaveStyle({
        padding: `${spacing.large}px ${spacing.medium}px`,
      });

      expect(container.firstChild).toMatchSnapshot();
    });

    it('the default version with v3Theme', () => {
      const { container } = render(
        <ThemeProvider theme={v3theme}>
          <Accordion title="Default">
            <Accordion.Content>
              <Text>Content</Text>
            </Accordion.Content>
          </Accordion>
        </ThemeProvider>,
      );

      expect(screen.getByText('Default')).toBeInTheDocument();

      expect(screen.getByText('Default')).toHaveStyle({
        fontSize: `${fontSizes.medium}px`,
        fontWeight: fontWeights.bold,
      });

      expect(screen.getByRole('button')).toHaveStyle({
        padding: `${spacing.large}px ${spacing.medium}px`,
      });

      expect(container.firstChild).toMatchSnapshot();
    });

    it('the small version', () => {
      const { container } = render(
        <ThemeProvider>
          <Accordion.Small title="Small">
            <Accordion.Content>
              <Text>Content</Text>
            </Accordion.Content>
          </Accordion.Small>
        </ThemeProvider>,
      );

      expect(screen.getByText('Small')).toBeInTheDocument();

      expect(screen.getByText('Small')).toHaveStyle({
        fontSize: `${fontSizes.small}px`,
        fontWeight: fontWeights.regular,
      });

      expect(screen.getByRole('button')).toHaveStyle({
        padding: `${spacing.small}px ${spacing.medium}px`,
      });

      expect(container.firstChild).toMatchSnapshot();
    });

    it('the default version with subtitle', () => {
      const { container } = render(
        <ThemeProvider>
          <Accordion title="Default" subtitle="default with subtitle">
            <Accordion.Content>
              <Text>Content</Text>
            </Accordion.Content>
          </Accordion>
        </ThemeProvider>,
      );

      expect(screen.getByText('Default')).toBeInTheDocument();
      expect(screen.getByText('default with subtitle')).toBeInTheDocument();

      expect(screen.getByText('Default')).toHaveStyle({
        fontSize: `${fontSizes.medium}px`,
        fontWeight: fontWeights.medium,
      });

      expect(screen.getByRole('button')).toHaveStyle({
        padding: `${spacing.large}px ${spacing.medium}px`,
      });

      expect(container.firstChild).toMatchSnapshot();
    });

    it('the small version with subtitle', () => {
      const { container } = render(
        <ThemeProvider>
          <Accordion.Small title="Small" subtitle="small with subtitle">
            <Accordion.Content>
              <Text>Content</Text>
            </Accordion.Content>
          </Accordion.Small>
        </ThemeProvider>,
      );

      expect(screen.getByText('Small')).toBeInTheDocument();
      expect(screen.getByText('small with subtitle')).toBeInTheDocument();

      expect(screen.getByText('Small')).toHaveStyle({
        fontSize: `${fontSizes.small}px`,
        fontWeight: fontWeights.regular,
      });

      expect(screen.getByRole('button')).toHaveStyle({
        padding: `${spacing.small}px ${spacing.medium}px`,
      });
      expect(container.firstChild).toMatchSnapshot();
    });

    it('the default version without side padding', () => {
      const { container } = render(
        <ThemeProvider>
          <Accordion title="Default" expanded hasHorizontalPadding={false}>
            <Accordion.Content>
              <Text>Content</Text>
            </Accordion.Content>
          </Accordion>
        </ThemeProvider>,
      );

      expect(screen.getByRole('button')).toHaveStyle({
        padding: `${spacing.large}px ${spacing.zero}px`,
      });

      expect(container.firstChild).toMatchSnapshot();
    });

    it('the small version without side padding', () => {
      const { container } = render(
        <ThemeProvider>
          <Accordion.Small title="Small" hasHorizontalPadding={false}>
            <Accordion.Content>
              <Text>Content</Text>
            </Accordion.Content>
          </Accordion.Small>
        </ThemeProvider>,
      );

      expect(screen.getByRole('button')).toHaveStyle({
        padding: `${spacing.small}px ${spacing.zero}px`,
      });

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should render the accordion summary', () => {
    render(
      <ThemeProvider>
        <Accordion title="Title" subtitle="Subtitle">
          <Accordion.Summary>
            <Text>Summary</Text>
          </Accordion.Summary>
          <Accordion.Content>
            <Text>Content</Text>
          </Accordion.Content>
        </Accordion>
      </ThemeProvider>,
    );

    expect(screen.getByText('Summary')).toBeInTheDocument();
    expect(screen.queryByText('Title')).not.toBeInTheDocument();
    expect(screen.queryByText('Subtitle')).not.toBeInTheDocument();
  });

  it('should have className when passed as props', () => {
    render(
      <ThemeProvider>
        <Accordion title="Title">
          <Accordion.Content
            className="className-as-props"
            data-testid="accordion-content"
          >
            <Text>Content</Text>

            <Button small inverted>
              Small button
            </Button>
          </Accordion.Content>
        </Accordion>
      </ThemeProvider>,
    );

    expect(screen.getByTestId('accordion-content')).toHaveClass(
      'className-as-props',
    );
  });
});
