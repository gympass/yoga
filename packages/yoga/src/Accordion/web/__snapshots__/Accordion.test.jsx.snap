// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should correctly render the default version 1`] = `
.c7 {
  padding: 0 20px 20px;
}

.c0 {
  background-color: #FFFFFF;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c0 hr {
  bottom: 0;
  left: 0;
  margin: 0;
  position: absolute;
  z-index: -1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 24px 20px;
}

.c1:focus-visible {
  background: #F5F5FA;
}

.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: start;
  gap: 0px;
  margin: 8px 0;
}

.c6 {
  height: auto;
  max-height: 9999px;
  overflow: hidden;
  -webkit-transition: max-height 200ms ease-in-out;
  transition: max-height 200ms ease-in-out;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
}

.c5 svg {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  -webkit-transition: all 200ms ease-out 0s;
  transition: all 200ms ease-out 0s;
  fill: #D8385E;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  line-height: 24px;
}

.c8 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-size: 16px;
}

.c9 {
  width: 100%;
  border-width: 0px;
  border-left-width: 0px;
  border-bottom-width: 1px;
  border-color: #D7D7E0;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Default
      </p>
      <p
        class="c4"
        color="deep"
      />
    </div>
    <div
      class="c5"
    >
      <svg
        height="24"
        width="24"
      />
    </div>
  </button>
  <div
    class="c6"
  >
    <div
      class="c7"
    >
      <p
        class="c8"
        font-size="medium"
      >
        Content
      </p>
    </div>
  </div>
  <hr
    class="c9"
  />
</div>
`;

exports[`<Accordion /> should correctly render the default version with subtitle 1`] = `
.c8 {
  padding: 0 20px 20px;
}

.c0 {
  background-color: #FFFFFF;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c0 hr {
  bottom: 0;
  left: 0;
  margin: 0;
  position: absolute;
  z-index: -1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 24px 20px;
}

.c1:focus-visible {
  background: #F5F5FA;
}

.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: start;
  gap: 4px;
  margin: 0;
}

.c11 .c7 {
  padding-bottom: 16px;
}

.c6 {
  height: auto;
  max-height: 0;
  overflow: hidden;
  -webkit-transition: max-height 200ms ease-in-out;
  transition: max-height 200ms ease-in-out;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
}

.c5 svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 200ms ease-out 0s;
  transition: all 200ms ease-out 0s;
  fill: #D8385E;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  line-height: 24px;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-size: 16px;
}

.c10 {
  width: 100%;
  border-width: 0px;
  border-left-width: 0px;
  border-bottom-width: 1px;
  border-color: #D7D7E0;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Default
      </p>
      <p
        class="c4"
        color="deep"
      >
        default with subtitle
      </p>
    </div>
    <div
      class="c5"
    >
      <svg
        height="24"
        width="24"
      />
    </div>
  </button>
  <div
    class="c6"
  >
    <div
      class="c7 c8"
    >
      <p
        class="c9"
        font-size="medium"
      >
        Content
      </p>
    </div>
  </div>
  <hr
    class="c10"
  />
</div>
`;

exports[`<Accordion /> should correctly render the default version without side padding 1`] = `
.c8 {
  padding: 0 20px 20px;
}

.c0 {
  background-color: #FFFFFF;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c0 hr {
  bottom: 0;
  left: 0;
  margin: 0;
  position: absolute;
  z-index: -1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 24px 0px;
}

.c1:focus-visible {
  background: #F5F5FA;
}

.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: start;
  gap: 0px;
  margin: 8px 0;
}

.c11 .c7 {
  padding-bottom: 16px;
}

.c6 {
  height: auto;
  max-height: 9999px;
  overflow: hidden;
  -webkit-transition: max-height 200ms ease-in-out;
  transition: max-height 200ms ease-in-out;
}

.c6 .c7 {
  padding-left: 0;
  padding-right: 0;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
}

.c5 svg {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  -webkit-transition: all 200ms ease-out 0s;
  transition: all 200ms ease-out 0s;
  fill: #D8385E;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  line-height: 24px;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-size: 16px;
}

.c10 {
  width: 100%;
  border-width: 0px;
  border-left-width: 0px;
  border-bottom-width: 1px;
  border-color: #D7D7E0;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Default
      </p>
      <p
        class="c4"
        color="deep"
      />
    </div>
    <div
      class="c5"
    >
      <svg
        height="24"
        width="24"
      />
    </div>
  </button>
  <div
    class="c6"
  >
    <div
      class="c7 c8"
    >
      <p
        class="c9"
        font-size="medium"
      >
        Content
      </p>
    </div>
  </div>
  <hr
    class="c10"
  />
</div>
`;

exports[`<Accordion /> should correctly render the small version 1`] = `
.c8 {
  padding: 0 20px 20px;
}

.c0 {
  background-color: #FFFFFF;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c0 hr {
  bottom: 0;
  left: 0;
  margin: 0;
  position: absolute;
  z-index: -1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 16px 20px;
}

.c1:focus-visible {
  background: #F5F5FA;
}

.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: start;
  gap: 0px;
  margin: 8px 0;
}

.c6 {
  height: auto;
  max-height: 0;
  overflow: hidden;
  -webkit-transition: max-height 200ms ease-in-out;
  transition: max-height 200ms ease-in-out;
}

.c6 .c7 {
  padding-bottom: 16px;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
}

.c5 svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 200ms ease-out 0s;
  transition: all 200ms ease-out 0s;
  fill: #D8385E;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-size: 16px;
}

.c10 {
  width: 100%;
  border-width: 0px;
  border-left-width: 0px;
  border-bottom-width: 1px;
  border-color: #D7D7E0;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Small
      </p>
      <p
        class="c4"
        color="deep"
      />
    </div>
    <div
      class="c5"
    >
      <svg
        height="24"
        width="24"
      />
    </div>
  </button>
  <div
    class="c6"
  >
    <div
      class="c7 c8"
    >
      <p
        class="c9"
        font-size="medium"
      >
        Content
      </p>
    </div>
  </div>
  <hr
    class="c10"
  />
</div>
`;

exports[`<Accordion /> should correctly render the small version with subtitle 1`] = `
.c8 {
  padding: 0 20px 20px;
}

.c0 {
  background-color: #FFFFFF;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c0 hr {
  bottom: 0;
  left: 0;
  margin: 0;
  position: absolute;
  z-index: -1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 16px 20px;
}

.c1:focus-visible {
  background: #F5F5FA;
}

.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: start;
  gap: 4px;
  margin: 0;
}

.c6 {
  height: auto;
  max-height: 0;
  overflow: hidden;
  -webkit-transition: max-height 200ms ease-in-out;
  transition: max-height 200ms ease-in-out;
}

.c6 .c7 {
  padding-bottom: 16px;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
}

.c5 svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 200ms ease-out 0s;
  transition: all 200ms ease-out 0s;
  fill: #D8385E;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-size: 16px;
}

.c10 {
  width: 100%;
  border-width: 0px;
  border-left-width: 0px;
  border-bottom-width: 1px;
  border-color: #D7D7E0;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Small
      </p>
      <p
        class="c4"
        color="deep"
      >
        small with subtitle
      </p>
    </div>
    <div
      class="c5"
    >
      <svg
        height="24"
        width="24"
      />
    </div>
  </button>
  <div
    class="c6"
  >
    <div
      class="c7 c8"
    >
      <p
        class="c9"
        font-size="medium"
      >
        Content
      </p>
    </div>
  </div>
  <hr
    class="c10"
  />
</div>
`;

exports[`<Accordion /> should correctly render the small version without side padding 1`] = `
.c8 {
  padding: 0 20px 20px;
}

.c0 {
  background-color: #FFFFFF;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c0 hr {
  bottom: 0;
  left: 0;
  margin: 0;
  position: absolute;
  z-index: -1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 16px 0px;
}

.c1:focus-visible {
  background: #F5F5FA;
}

.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: start;
  gap: 0px;
  margin: 8px 0;
}

.c11 .c7 {
  padding-bottom: 16px;
}

.c12 .c7 {
  padding-left: 0;
  padding-right: 0;
}

.c6 {
  height: auto;
  max-height: 0;
  overflow: hidden;
  -webkit-transition: max-height 200ms ease-in-out;
  transition: max-height 200ms ease-in-out;
}

.c6 .c7 {
  padding-bottom: 16px;
}

.c6 .c7 {
  padding-left: 0;
  padding-right: 0;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
}

.c5 svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 200ms ease-out 0s;
  transition: all 200ms ease-out 0s;
  fill: #D8385E;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-size: 16px;
}

.c10 {
  width: 100%;
  border-width: 0px;
  border-left-width: 0px;
  border-bottom-width: 1px;
  border-color: #D7D7E0;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Small
      </p>
      <p
        class="c4"
        color="deep"
      />
    </div>
    <div
      class="c5"
    >
      <svg
        height="24"
        width="24"
      />
    </div>
  </button>
  <div
    class="c6"
  >
    <div
      class="c7 c8"
    >
      <p
        class="c9"
        font-size="medium"
      >
        Content
      </p>
    </div>
  </div>
  <hr
    class="c10"
  />
</div>
`;
